### PR TITLE
Refactored storage actions into the Storage module

### PR DESCRIPTION
**Pull request by UtkarshBhatthere**

### CODE REVIEW

File: src/storage.py [hunk 1]
```diff
@@ -150,6 +219,8 @@ def remove_osd(self, osd_num: int, force: bool = False):
             microceph.remove_disk_cmd(osd_num, force)
             # if no errors while removing OSD, clean stale osd records.
             self._clean_stale_osd_data()
+            # Request charm to update all clients.
+            self.on.unit_storage_changed.emit()
         except CalledProcessError as e:
             if force:
```
Comment (lmlg) [inline src/storage.py:225]: shouldn't we notify of changes to clients here as well?

**Review by lmlg — COMMENTED** [review]
lgtm, just a quick question.

File: src/storage.py (hunk 1 above)
Comment (UtkarshBhatthere) [reply src/storage.py:225]: ack, nice catch. have added.

**Review by lmlg — DISMISSED** [review]
lgtm, good stuff.

**Review by sabaini — APPROVED** [review]
lgtm

**Pull request merged**